internal/integration/provision/upgrade.go

```diff
@@ -54,7 +54,7 @@ type upgradeSpec struct {
 
 const (
 	talos03Version = "v0.3.2-1-g71ac6696"
-	talos04Version = "v0.4.0-alpha.5"
+	talos04Version = "v0.4.0-alpha.5-19-g8913d9df"
```

Comment:
Should we use `latest` here?

Reply:
this is source version to upgrade from, so I think we should use the earliest version upgrade works from?

Reply:
well, actually it's both destination and source, it's source for 0.4 -> latest and destination for 0.3 -> 0.4 probably I should split them ?

Reply:
for 0.4 -> latest upgrade, it makes sense to use the earliest version which works for upgrades, and for 0.3 -> 0.4 it probably makes sense to use the latest.

the problem with :latest is that we don't know the exact version so it's hard to assert on upgrade correctness, we assert that Talos version after upgrade matches expected

Reply:
👍